test/connect.js

```diff
@@ -118,4 +118,15 @@ describe('Basic Connectivity', function() {
     }, 100 * 2);
   });
 
+
+  it('should add a new cluster server', function(done){
+    var natsServers = [uri]
+    var nc = NATS.connect({servers: natsServers});
+    nc.on('connect', function(client) {
+      client.addServer('nats://localhost:22223');
+      client.servers.length.should.equal(2);
```

Comment:
I think that sounds reasonable. Only hesitation I have is that if we end up with an order that hs me picking the same server on a disconnect for the subsequent re-connect. But I think that is something we can live with. So just call to shuffle() if opts.NoRandomize is false.